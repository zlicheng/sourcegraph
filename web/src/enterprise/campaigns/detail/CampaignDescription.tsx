import React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { Timestamp } from '../../../components/time/Timestamp'
import { Markdown } from '../../../../../shared/src/components/Markdown'
import H from 'history'
import { renderMarkdown } from '../../../../../shared/src/util/markdown'
import { Link } from 'react-router-dom'

interface Props {
    campaign: Pick<GQL.ICampaign, 'description' | 'updatedAt'> & {
        author: Pick<GQL.ICampaign['author'], 'username' | 'url'>
    }
    history: H.History
    className?: string
}

export const CampaignDescription: React.FunctionComponent<Props> = ({ campaign, history, className = '' }) => (
    <div className={`card ${className}`}>
        <div className="card-header">
            <Link to={campaign.author.url}>
                <strong>{campaign.author.username}</strong>
            </Link>{' '}
            <span className="text-muted">
                <Timestamp date={campaign.updatedAt} />
            </span>
        </div>
        <div className="card-body">
            <Markdown
                dangerousInnerHTML={renderMarkdown(campaign.description || '_No description_')}
                history={history}
            />
        </div>
    </div>
)
