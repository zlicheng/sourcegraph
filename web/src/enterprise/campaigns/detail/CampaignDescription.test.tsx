import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { CampaignDescription } from './CampaignDescription'
import * as H from 'history'

describe('CampaignDescription', () => {
    test('', () =>
        expect(
            createRenderer().render(
                <CampaignDescription
                    campaign={{
                        author: { avatarURL: null, username: 'alice' },
                        createdAt: '2020-01-01',
                        description: '**a** b',
                    }}
                    history={H.createMemoryHistory()}
                />
            )
        ).toMatchSnapshot())
})
