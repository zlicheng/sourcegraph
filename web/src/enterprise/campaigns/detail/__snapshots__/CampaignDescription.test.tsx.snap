// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignDescription  1`] = `
<div
  className="card mt-2"
>
  <div
    className="card-header"
  >
    <strong>
      <UserAvatar
        className="icon-inline"
        user={Object {}}
      />
       
    </strong>
     
    started 
    <Timestamp
      date="2020-01-01"
    />
  </div>
  <div
    className="card-body"
  >
    <Markdown
      dangerousInnerHTML="<p><strong>a</strong> b</p>
"
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": undefined,
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": undefined,
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    />
  </div>
</div>
`;
